<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>400</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Galactic Calculator</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout" columnstretch="0">
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item row="1" column="0">
       <widget class="QTabWidget" name="mainTabWidget">
        <property name="currentIndex">
         <number>3</number>
        </property>
        <widget class="QWidget" name="unitExcahngeTab">
         <attribute name="title">
          <string>unit exchange</string>
         </attribute>
        </widget>
        <widget class="QWidget" name="computeTab">
         <attribute name="title">
          <string>Calculation</string>
         </attribute>
        </widget>
        <widget class="QWidget" name="routeTab">
         <attribute name="title">
          <string>Directions</string>
         </attribute>
        </widget>
        <widget class="QWidget" name="dbTab">
         <attribute name="title">
          <string>Database transactions</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>1</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTabWidget" name="dbTabWidget">
            <property name="statusTip">
             <string/>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="listDataTab">
             <attribute name="title">
              <string>List database</string>
             </attribute>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QWidget" name="listShipParentWidget" native="true">
                <layout class="QGridLayout" name="gridLayout_5">
                 <property name="rightMargin">
                  <number>14</number>
                 </property>
                 <item row="0" column="0">
                  <widget class="QLabel" name="listShipsLabel">
                   <property name="font">
                    <font>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string>List spaceships</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLineEdit" name="shipSearchTextBox"/>
                 </item>
                 <item row="1" column="1">
                  <spacer name="ShipHorizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Maximum</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>10</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item row="1" column="2">
                  <widget class="QPushButton" name="searchShipPushButton">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>List</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0" colspan="3">
                  <widget class="QTableWidget" name="shipTableWidget">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                     <horstretch>1</horstretch>
                     <verstretch>1</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="font">
                    <font>
                     <strikeout>false</strikeout>
                    </font>
                   </property>
                   <property name="editTriggers">
                    <set>QAbstractItemView::NoEditTriggers</set>
                   </property>
                   <property name="alternatingRowColors">
                    <bool>true</bool>
                   </property>
                   <property name="textElideMode">
                    <enum>Qt::ElideMiddle</enum>
                   </property>
                   <property name="sortingEnabled">
                    <bool>true</bool>
                   </property>
                   <property name="wordWrap">
                    <bool>true</bool>
                   </property>
                   <property name="columnCount">
                    <number>5</number>
                   </property>
                   <attribute name="horizontalHeaderCascadingSectionResizes">
                    <bool>true</bool>
                   </attribute>
                   <attribute name="horizontalHeaderDefaultSectionSize">
                    <number>85</number>
                   </attribute>
                   <attribute name="horizontalHeaderStretchLastSection">
                    <bool>true</bool>
                   </attribute>
                   <attribute name="verticalHeaderCascadingSectionResizes">
                    <bool>true</bool>
                   </attribute>
                   <column>
                    <property name="text">
                     <string>Name</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Velocity</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Velocity unit</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Acceleration</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Acceleration unit</string>
                    </property>
                   </column>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="listRouteParentWidget" native="true">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <layout class="QGridLayout" name="gridLayout_4" rowstretch="0,0,0">
                 <property name="sizeConstraint">
                  <enum>QLayout::SetMaximumSize</enum>
                 </property>
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>9</number>
                 </property>
                 <property name="rightMargin">
                  <number>6</number>
                 </property>
                 <property name="bottomMargin">
                  <number>9</number>
                 </property>
                 <property name="verticalSpacing">
                  <number>6</number>
                 </property>
                 <item row="1" column="2">
                  <widget class="QPushButton" name="listDBPushButton">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>100</width>
                     <height>20</height>
                    </size>
                   </property>
                   <property name="statusTip">
                    <string>List the routes and apply the filter.</string>
                   </property>
                   <property name="text">
                    <string>List</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <spacer name="horizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Fixed</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>10</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLineEdit" name="searchTextBox">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="statusTip">
                    <string>Type a keyword and search in routes by name</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0" colspan="3">
                  <widget class="QTableWidget" name="tableWidget">
                   <property name="enabled">
                    <bool>true</bool>
                   </property>
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                     <horstretch>1</horstretch>
                     <verstretch>1</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>320</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="statusTip">
                    <string>Table of routes</string>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="midLineWidth">
                    <number>1</number>
                   </property>
                   <property name="editTriggers">
                    <set>QAbstractItemView::NoEditTriggers</set>
                   </property>
                   <property name="alternatingRowColors">
                    <bool>true</bool>
                   </property>
                   <property name="gridStyle">
                    <enum>Qt::CustomDashLine</enum>
                   </property>
                   <property name="sortingEnabled">
                    <bool>true</bool>
                   </property>
                   <property name="rowCount">
                    <number>0</number>
                   </property>
                   <property name="columnCount">
                    <number>3</number>
                   </property>
                   <attribute name="verticalHeaderCascadingSectionResizes">
                    <bool>true</bool>
                   </attribute>
                   <attribute name="verticalHeaderShowSortIndicator" stdset="0">
                    <bool>false</bool>
                   </attribute>
                   <column>
                    <property name="text">
                     <string>Route</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Distance</string>
                    </property>
                   </column>
                   <column>
                    <property name="text">
                     <string>Unit</string>
                    </property>
                   </column>
                  </widget>
                 </item>
                 <item row="0" column="0">
                  <widget class="QLabel" name="listRoutesLabel">
                   <property name="font">
                    <font>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="text">
                    <string>List routes</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="newDataTab">
             <attribute name="title">
              <string>Add data</string>
             </attribute>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <layout class="QFormLayout" name="addRouteFormLayout">
                <property name="sizeConstraint">
                 <enum>QLayout::SetDefaultConstraint</enum>
                </property>
                <property name="fieldGrowthPolicy">
                 <enum>QFormLayout::FieldsStayAtSizeHint</enum>
                </property>
                <property name="horizontalSpacing">
                 <number>24</number>
                </property>
                <property name="verticalSpacing">
                 <number>6</number>
                </property>
                <property name="leftMargin">
                 <number>6</number>
                </property>
                <property name="topMargin">
                 <number>6</number>
                </property>
                <property name="rightMargin">
                 <number>6</number>
                </property>
                <item row="1" column="0">
                 <widget class="QLabel" name="routeNameLabel">
                  <property name="text">
                   <string>Name</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="distanceLabel">
                  <property name="text">
                   <string>Distance</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QDoubleSpinBox" name="quantityInput">
                  <property name="statusTip">
                   <string>Distance of the route.</string>
                  </property>
                  <property name="maximum">
                   <double>999999.989999999990687</double>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QComboBox" name="distanceUnitcomboBox">
                  <property name="statusTip">
                   <string>Unit of the distance</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="0">
                 <widget class="QLabel" name="unitLabel">
                  <property name="text">
                   <string>Unit</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="0">
                 <widget class="QPushButton" name="addNewRootButton">
                  <property name="statusTip">
                   <string>Add the new route</string>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0" colspan="2">
                 <widget class="QLabel" name="addNewRouteLabel">
                  <property name="font">
                   <font>
                    <weight>75</weight>
                    <bold>true</bold>
                   </font>
                  </property>
                  <property name="text">
                   <string>Add a new Route</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="nameText">
                  <property name="statusTip">
                   <string>Name of the route. You must type this.</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QFormLayout" name="addShipFormLayout">
                <property name="sizeConstraint">
                 <enum>QLayout::SetMaximumSize</enum>
                </property>
                <property name="fieldGrowthPolicy">
                 <enum>QFormLayout::ExpandingFieldsGrow</enum>
                </property>
                <property name="horizontalSpacing">
                 <number>24</number>
                </property>
                <property name="verticalSpacing">
                 <number>6</number>
                </property>
                <property name="leftMargin">
                 <number>6</number>
                </property>
                <property name="topMargin">
                 <number>6</number>
                </property>
                <property name="rightMargin">
                 <number>6</number>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="addNewSpaceShipLabel">
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="font">
                   <font>
                    <weight>75</weight>
                    <bold>true</bold>
                   </font>
                  </property>
                  <property name="text">
                   <string>Add a new spaceship</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="shipNameLabel">
                  <property name="text">
                   <string>Name</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLineEdit" name="shipNameTextLineEdit"/>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="velocityLabel">
                  <property name="text">
                   <string>Velocity</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="1">
                 <widget class="QDoubleSpinBox" name="velocityInput"/>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="velocityUnitLabel">
                  <property name="text">
                   <string>Velocity unit</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <widget class="QComboBox" name="VeloctiyUnitComboBox"/>
                </item>
                <item row="5" column="0">
                 <widget class="QLabel" name="accelerationLabel">
                  <property name="text">
                   <string>Acceleration</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QDoubleSpinBox" name="accelerationInput"/>
                </item>
                <item row="6" column="0">
                 <widget class="QLabel" name="accelerationUnitLabel">
                  <property name="text">
                   <string>Acceleration unit</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="1">
                 <widget class="QComboBox" name="accelerationUnitCOmboBox"/>
                </item>
                <item row="7" column="0">
                 <widget class="QPushButton" name="addNewSpaceShipPushButton">
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAboutQt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Import</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAboutQt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
